src/mlpack/methods/ann/layer/noisylinear_impl.hpp

```diff
@@ -48,6 +48,48 @@ NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
   biasEpsilon.set_size(outSize, 1);
 }
 
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
+  NoisyLinear&& layer) :
+    inSize(std::move(layer.inSize)),
+    outSize(std::move(layer.outSize)),
+    weights(std::move(layer.weights))
+{
+  layer.inSize = 0;
+  layer.outSize = 0;
+  layer.weights = nullptr;
+  Reset();
+}
+
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>&
+NoisyLinear<InputDataType, OutputDataType>::operator=(NoisyLinear const& layer)
+{
+  if(this != &layer) {
+    inSize = layer.inSize;
+    outSize = layer.outSize;
+    weights = layer.weights;
+    Reset();
+  }
```

Comment:
```suggestion
  if( this != &layer)
  {
    inSize = layer.inSize;
    outSize = layer.outSize;
    weights = layer.weights;
    Reset();
  }
```

Just to match with the coding style - https://github.com/mlpack/mlpack/wiki/DesignGuidelines.